src/coreclr/jit/unwindamd64.cpp

```diff
@@ -423,7 +423,20 @@ void Compiler::unwindSaveRegWindows(regNumber reg, unsigned offset)
             code             = (UNWIND_CODE*)&func->unwindCodes[func->unwindCodeSlot -= sizeof(UNWIND_CODE)];
             code->UnwindOp   = (genIsValidFloatReg(reg)) ? UWOP_SAVE_XMM128_FAR : UWOP_SAVE_NONVOL_FAR;
         }
-        code->OpInfo          = (BYTE)(genIsValidFloatReg(reg) ? reg - XMMBASE : reg);
+        unsigned unwindRegNum;
+        if (genIsValidFloatReg(reg))
+        {
+            unwindRegNum = reg - XMMBASE;
+        }
+        else
+        {
+            assert(genIsValidIntReg(reg));
+            unwindRegNum = reg;
+        }
+        assert(unwindRegNum <= 15);
```

Comment:
Is this correct?

We have 32 float registers with AVX512 and can have up to 32 general-purpose registers with APX.

If we're relying on something like this only being non-volatile registers, then I'd think we want a comment explaining that.

Reply:
`OpInfo` only has 4 bits, so it can only fit up to 15, so if there were ever callee-saves with a higher index it would also need some other unwind code.